R/annotate.R

```diff
@@ -0,0 +1,125 @@
+# annotate.R
+
+#' Produce annotated gene graph where vertices have scores
+#' attributed to genes.
+#'
+#' \code{annotateGraph} takes as inputs a gene heat hashtable and a gene graph,
+#' and annotates the graph with gene heats that have been calculated by the
+#' scoring function. It then returns an AGG (annotated gene graph).
+#'
+#' @param gH List of genes and their associated score (mutational frequncy of
+#'           mutSigCV value)
+#' @param gG gene graph
+#' @param silent Controls whether output to console should be suppressed. FALSE
+#'   by default.
+#' @param writeLog Controls whether writing the result to the global logfile is
+#'   enabled. TRUE by default.
+#' @return AGG object with annotated vertices
+#'
+#' @examples
+# \dontrun{ annotateGraph(gH, gG) }
+#'
+#' @export
+annotateGraph <- function(gH, gG, silent = FALSE, writeLog = TRUE) {
+
+    gHName <- deparse(substitute(gH))
+    gGName <- deparse(substitute(gG))
+
+    # check for params - that they are not missing
+    if (is.null(gH)) {
+        stop(sprintf("Object \"%s\" is NULL.",
+                     gHName))
+    }
+
```

Comment:
We should use ```.checkArgs()``` here. I'll add this later. The Way is to build a prototype object in the package globals, then use ```.checkArgs(gH like = getOption("rete.gHprototype"))```